src/app/clusters/ota-server/ota-server.cpp

```diff
@@ -82,5 +123,24 @@ bool emberAfOtaSoftwareUpdateServerClusterQueryImageCallback(
     /* TYPE WARNING: array array defaults to */ uint8_t * protocolsSupported, uint8_t * location, uint8_t clientCanConsent,
     chip::ByteSpan metadataForServer)
 {
-    return false;
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    ChipLogDetail(Zcl, "OTA Server received QueryImage");
+
+    const uint8_t locationLen = emberAfStringLength(location);
+    if ((locationLen != kLocationParamLength) || (metadataForServer.size() > kMaxMetadataLen))
+    {
+        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_FIELD);
+    }
+
+    chip::ByteSpan locationSpan(location, locationLen);
+
+    status = OTAServerDelegate::HandleQueryImage(vendorId, productId, imageType, hardwareVersion, currentVersion,
```

Comment:
The API should allow different implementations to be used, which may embed different state-keeping. Suggest making the OTAServerDelegate a virtual interface.

Reply:
Agreed. I made the Delegate methods virtual and added a function for setting the Delegate for the ember callbacks to use